cwms-http-client/src/main/java/mil/army/usace/hec/cwms/http/client/CwmsHttpLogger.java

```diff
@@ -0,0 +1,44 @@
+package mil.army.usace.hec.cwms.http.client;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.util.Map;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import okhttp3.logging.HttpLoggingInterceptor;
+
+final class CwmsHttpLogger implements HttpLoggingInterceptor.Logger {
+
+    private static final String ACCESS_TOKEN = "access_token";
+    private static final String REFRESH_TOKEN = "refresh_token";
+    private static final Logger LOGGER = Logger.getLogger(CwmsHttpLogger.class.getName());
+    private static final HttpLoggingInterceptor.Logger DELEGATE = HttpLoggingInterceptor.Logger.DEFAULT;
+
+    @Override
+    public void log(String s) {
+        String lowerCaseS = s.toLowerCase();
+        if (!lowerCaseS.contains(ACCESS_TOKEN.toLowerCase()) && !lowerCaseS.contains(REFRESH_TOKEN.toLowerCase())) {
+            DELEGATE.log(s);
+        } else {
+            try {
+                s = redactOAuth2Tokens(s);
+                DELEGATE.log(s);
+            } catch (JsonProcessingException e) {
+                LOGGER.log(Level.INFO, () -> "Could not parse logged JSON. Redacting log containing token(s) for security reasons.");
+            }
+        }
+    }
+
+    private String redactOAuth2Tokens(String s) throws JsonProcessingException {
+        ObjectMapper om = new ObjectMapper();
+        Map<String, String> jsonMap = om.readValue(s, Map.class);
```

Comment:
This only traverses the root, not all child elements. I recommend either not logging at all if it contains the tokens or traversing the entire JSON tree.

Reply:
Replaced this logic with redacting string completely